DatadogRUM/Sources/Integrations/TelemetryReceiver.swift

```diff
@@ -255,11 +259,18 @@ internal final class TelemetryReceiver: FeatureMessageReceiver {
             let sessionIDOverride: String? = attributes.removeValue(forKey: SDKMetricFields.sessionIDOverrideKey)?.dd.decode()
             let sessionID = sessionIDOverride ?? rum?.sessionID
 
+            // Calculates the composition of sample rates. The metric can have up to 3 layers of sampling.
+            var effectiveSampleRate = metric.sampleRate.composed(with: self.sampler.samplingRate)
+            if let headSampleRate = metric.attributes[MethodCalledMetric.headSampleRate] as? SampleRate {
+                effectiveSampleRate = effectiveSampleRate.composed(with: headSampleRate)
+            }
+
             let event = TelemetryDebugEvent(
                 dd: .init(),
                 action: rum?.userActionID.map { .init(id: $0) },
                 application: rum.map { .init(id: $0.applicationID) },
                 date: date.addingTimeInterval(context.serverTimeOffset).timeIntervalSince1970.toInt64Milliseconds,
+                effectiveSampleRate: Int64(effectiveSampleRate),
```

Comment:
blocking/ This can overflow, especially that `telemetrySampleRate` is controlled by the user. We should use [`Int64(withNoOverflow:)`](https://github.com/DataDog/dd-sdk-ios/blob/a4c6a1cb49a22eec9f2d5dada8be160082533d35/DatadogInternal/Sources/Utils/SwiftExtensions.swift#L96-L99) instead.

Reply:
You're right. ty